app/src/components/BlinkoSettings/AiSetting/AuthenticationConfig.tsx

```diff
@@ -0,0 +1,343 @@
+import { Icon } from '@/components/Common/Iconify/icons';
+import { useTranslation } from 'react-i18next';
+import { useState, useEffect } from 'react';
+import { AuthenticationConfig } from '@/store/aiSettingStore';
+
+interface AuthenticationConfigProps {
+  authConfig?: AuthenticationConfig;
+  onAuthConfigChange: (config: AuthenticationConfig) => void;
+  provider?: string;
+}
+
+// Default authentication configurations for common providers
+const DEFAULT_AUTH_CONFIGS: { [key: string]: AuthenticationConfig } = {
+  openai: {
+    type: 'bearer',
+    headerName: 'Authorization'
+  },
+  anthropic: {
+    type: 'api-key',
+    headerName: 'x-api-key'
+  },
+  azure: {
+    type: 'api-key',
+    headerName: 'api-key'
+  },
+  google: {
+    type: 'api-key',
+    headerName: 'x-goog-api-key'
+  },
+  voyageai: {
+    type: 'api-key',
+    headerName: 'Authorization'
+  },
+  zhipuai: {
+    type: 'bearer',
+    headerName: 'Authorization'
+  },
+  ollama: {
+    type: 'none'
+  },
+  custom: {
+    type: 'bearer',
+    headerName: 'Authorization'
+  }
+};
+
+export default observer(function AuthenticationConfig({
+  authConfig,
+  onAuthConfigChange,
+  provider = 'custom'
+}: AuthenticationConfigProps) {
+  const { t } = useTranslation();
+  const [localConfig, setLocalConfig] = useState<AuthenticationConfig>(() => {
+    if (authConfig) {
+      return authConfig;
+    }
+
+    // Use default config for the provider type
+    return DEFAULT_AUTH_CONFIGS[provider] || DEFAULT_AUTH_CONFIGS.custom;
+  });
+
+  const [customHeaders, setCustomHeaders] = useState<{ [key: string]: string }>(
+    authConfig?.customHeaders || {}
+  );
+  const [customHeaderKey, setCustomHeaderKey] = useState('');
+  const [customHeaderValue, setCustomHeaderValue] = useState('');
+
+  useEffect(() => {
+    if (authConfig) {
+      setLocalConfig(authConfig);
+      setCustomHeaders(authConfig.customHeaders || {});
+    } else {
+      const defaultConfig = DEFAULT_AUTH_CONFIGS[provider] || DEFAULT_AUTH_CONFIGS.custom;
+      setLocalConfig(defaultConfig);
+      setCustomHeaders(defaultConfig.customHeaders || {});
+    }
+  }, [authConfig, provider]);
+
+  const updateConfig = (updates: Partial<AuthenticationConfig>) => {
+    const newConfig = { ...localConfig, ...updates };
+    setLocalConfig(newConfig);
+    onAuthConfigChange(newConfig);
+  };
+
+  const addCustomHeader = () => {
+    if (customHeaderKey.trim() && customHeaderValue.trim()) {
+      const newHeaders = { ...customHeaders, [customHeaderKey.trim()]: customHeaderValue.trim() };
+      setCustomHeaders(newHeaders);
+      updateConfig({ customHeaders: newHeaders });
+      setCustomHeaderKey('');
+      setCustomHeaderValue('');
+    }
+  };
+
+  const removeCustomHeader = (key: string) => {
+    const newHeaders = { ...customHeaders };
+    delete newHeaders[key];
+    setCustomHeaders(newHeaders);
+    updateConfig({ customHeaders: newHeaders });
+  };
+
+  const renderBearerConfig = () => (
```

Comment:
I learned that rendering by function could create problems when rerendering.
Instead, the best practice is to create a React component.
Simply like this:
```suggestion
  const BearerConfig = <div... 
```
To then use it with:
```
{localConfig.type === 'bearer' && <BearerConfig/>}
```